frontend/src/reducers/checklists.js

```diff
@@ -294,6 +310,118 @@ export default (state = defaultState, { type, meta, payload, error }) => {
         removeEntryError: error
       }
 
+    case ADD_COLLABORATOR_REQUEST:
+      return {
+        ...state,
+        creatingCollaborator: true,
+        createdCollaborator: false,
+        createCollaboratorError: null
+      }
+
+    case ADD_COLLABORATOR_SUCCESS:
+      return {
+        ...state,
+        collaboratorsByListId: {
+          ...state.collaboratorsByListId,
+          [meta.listId]: [
+            ...(state.collaboratorsByListId[meta.listId] || []),
+            {
+              title: meta.email,
+              ...payload
+            }
+          ]
+        },
+        creatingCollaborator: false,
+        createdCollaborator: true,
+        createCollaboratorError: null
+      }
+
+    case ADD_COLLABORATOR_FAILURE:
+      return {
+        ...state,
+        creatingCollaborator: false,
+        createdCollaborator: false,
+        createCollaboratorError: error
+      }
+
+    case REMOVE_COLLABORATOR_REQUEST:
+      return {
+        ...state,
+        removingCollaborator: true,
+        collaboratorRemoved: false,
+        collaboratorRemovalError: null
+      }
+
+    case REMOVE_COLLABORATOR_SUCCESS:
+      return {
+        ...state,
+        removingCollaborator: false,
```

Comment:
Why does this not result in the collaborator being removed from `collaboratorsByListId`?